### PR TITLE
Added INSurVeyor.

### PR DESCRIPTION
**Pull request by Mesh89**
Describe your pull request here

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

For members of the Bioconda project, the following command is also available:

<table>
  <tr>
    <td><code>@BiocondaBot please merge</code></td>
    <td>Upload built packages/containers and merge a PR. <br />Someone must approve a PR first! <br />This reduces CI build time by reusing built artifacts.</td>
  </tr>
</table>

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

File: recipes/insurveyor/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  sha256: 33c85157892d3256abc96bb2a9053f05da9dcf55befeed720e965577db0b78b5
+
+build:
+  number: 0
+
+requirements:
+  build:
+    - {{ compiler('cxx') }}
+  host:
+    - htslib ==1.13
```
Comment (bgruening) [inline recipes/insurveyor/meta.yaml:19]: Is it really necessary to pin htslib so strictly?

File: recipes/insurveyor/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  run:
+    - htslib ==1.13
```
Comment (bgruening) [inline recipes/insurveyor/meta.yaml:21]: you don't need to put htslib here, it has run_exports defined.

File: recipes/insurveyor/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    - pysam ==0.16.0.1
+    - pyfaidx ==0.5.9.1
+    - numpy
+
+test:
```
Comment (bgruening) [inline recipes/insurveyor/meta.yaml:26]: can you please include a test similar to `surveyor.py --help`

File: recipes/insurveyor/meta.yaml (hunk 1 above)
Comment (Mesh89) [reply recipes/insurveyor/meta.yaml:19]: The software will run with most recent versions, however the software was extensively tested with 1.13, and the results we will soon publish were produced with that version. I am afraid that if small changes are introduced in future versions of htslib, the results may be inconsistent. Do you see any drawbacks in pinning the release? Will it be deprecated in the future?

File: recipes/insurveyor/meta.yaml (hunk 3 above)
Comment (Mesh89) [reply recipes/insurveyor/meta.yaml:26]: Should I simply add "surveyor.py --help" as a test?

Comment (Mesh89) [reply recipes/insurveyor/meta.yaml:25]: Added to run_test.sh

**Comment by Mesh89** [comment]
@BiocondaBot please update

**Comment by dpryan79** [comment]
@BiocondaBot please fetch artifacts

File: recipes/insurveyor/meta.yaml (hunk 1 above)
Comment (dpryan79) [reply recipes/insurveyor/meta.yaml:19]: All other packages are built with 1.16, so it's impossible to install this in the same environment as them.

Comment (Mesh89) [reply recipes/insurveyor/meta.yaml:19]: Ok. I will test it with 1.16 and remove the constraint on 1.13 in the next release.

**Review by dpryan79 — APPROVED** [review]

**Comment by dpryan79** [comment]
@BiocondaBot please fetch artifacts

**Comment by dpryan79** [comment]
@BiocondaBot please fetch artifacts

**Comment by Mesh89** [comment]
@BiocondaBot please merge

**Pull request merged**